src/ring/middleware/session_timeout.clj

```diff
@@ -4,6 +4,40 @@
 (defn- current-time []
   (quot (System/currentTimeMillis) 1000))
 
+(defn- session-expired?
+  [request]
+  (let [session (:session request {})
+        end-time (or (::absolute-timeout session)
+                     (::idle-timeout session))]
+    (and end-time (< end-time (current-time)))))
+
+(defn- wrap-session-timeout
```

Comment:
I don't think we should factor this function out. Let's start with just repetition for the idle and absolute timeout middleware.